### PR TITLE
Add module for CVE-2015-1155 OS X Safari vulnerability

### PR DESCRIPTION
**Pull request by joevennix**
This adds the module for exploiting CVE-2015-1155 in Mac OS X Safari, which allows UXSS, theft of cookie database, theft of arbitrary files, and silent installation of Safari extensions. I put the .webarchive generation logic into its own mixin, so that my updates can be ported back to the `apple_safari_webarchive_uxss`. This will be done in a subsequent PR.

The actual exploit can be observed in `safari_file_url_navigation.rb#popup_html`. By abusing a cross-domain `history.pushState` function, we can navigate a popup window to an arbitrary `file:// URL`. By mounting an FTP network drive containing a malicious webarchive file, all the crazy privileges given to a normal (unquarantined) webarchive file can be achieved.

Apple fixed this bug in May, and also added X-Frame-Options to extensions.apple.com. I've updated the module to work around this.
#### Verification:
- [x] Exploit a vulnerable OSX Safari version (you have to click on the page once)
- [x] Cookie db is stolen
- [x] List of usernames is stolen
- [x] Any files in FILE_URLS are stolen
- [x] When INSTALL_EXTENSION is enabled, a Safari extension is silently installed

### CODE REVIEW

**Comment by wchen-r7** [comment]
@joevennix So I tested this on Safari 7.0.5. It didn't try to mount FTP. I also tried 8.0.5. That one mounted, but nothing stolen. Can you please tell me what version you've tested the module on? Thanks.

File: modules/auxiliary/gather/safari_file_url_navigation.rb [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    end
+  end
+
+  def on_request_uri(cli, req)
+    if req.method =~ /post/i
+      data_str = req.body.to_s
+      begin
+        data = JSON::parse(data_str || '')
+        file = record_data(data, cli)
+        send_response_html(cli, '')
```
Comment (wchen-r7) [inline modules/auxiliary/gather/safari_file_url_navigation.rb]: Just send_response

**Comment by joevennix** [comment]
I tested on 8.0.5, but tweaked some things before pushing to code review. I'll look at 8.0.5 again soon and see what went wrong.

**Comment by wchen-r7** [comment]
Ok cool

**Comment by joevennix** [comment]
@wchen-r7, I fixed some bugs in the mixin. My fix to avoid x-frame-options didn't work, as the extensions.apple.com window reference was being lost. The module should steal files, but it takes some moments (probably because the binary cookie file is being parsed in javascript, and the keychain is serialized into JSON).

**Comment by wchen-r7** [comment]
OK, I will try again. Thanks!

**Comment by wchen-r7** [comment]
Module works for me on my OSX box with Safari 7.0.5. I will merge this now. Thanks @joevennix

**Pull request merged**